app/views.py

```diff
@@ -0,0 +1,202 @@
+            flash("Login successful!", "success")
+            return redirect(url_for("home"))
+        else:
+            flash("Login failed. Check your email and password", "danger")
+    return render_template("login.html", form=form)
+
+
+@app.route("/request-reset-password", methods=["GET", "POST"])
+def request_reset_password():
+    if current_user.is_authenticated:
+        return redirect(url_for("home"))
+    form = UserRequestResetPasswordForm()
+    if request.method == "POST" and form.validate_on_submit():
+        user = User.query.filter_by(email=form.email.data).first()
+        if user:
+            encoded_jwt = jwt.encode(
+                {
+                    "user_id": user.id,
+                    "exp": datetime.now(tz=timezone.utc) + timedelta(minutes=10),
+                },
+                app.config["SECRET_KEY"],
+                algorithm="HS256",
+            )
+            send_email(form.email.data, encoded_jwt)
+        flash(
+            "If you have an account registered with this email, we have sent you a link to reset a password.",
+            "success",
+        )
+    return render_template("request_reset_password.html", form=form)
+
+
+@app.route("/reset-password", methods=["GET", "POST"])
+def reset_password():
+    if current_user.is_authenticated:
+        return redirect(url_for("home"))
+    token = request.args.get("token", "", type=str)
+    try:
+        payload = jwt.decode(token, app.config["SECRET_KEY"], algorithms="HS256")
+        user = User.query.get(payload["user_id"])
+        form = UserResetPasswordForm()
+        if user and request.method == "POST" and form.validate_on_submit():
+            encrypted_password = bcrypt.generate_password_hash(
+                form.password.data
+            ).decode("utf-8")
+            user.password = encrypted_password
+            db.session.add(user)
+            db.session.commit()
+            flash("Password changed! You can now login.", "success")
+            return redirect(url_for("login"))
+        return render_template("reset_password.html", form=form, token=token)
+    except jwt.InvalidSignatureError:
+        flash("Error or link no longer valid", "danger")
+        return redirect(url_for("login"))
+    except jwt.ExpiredSignatureError:
+        flash("Error or link no longer valid", "danger")
+        return redirect(url_for("login"))
+
+
+@app.route("/logout", methods=["GET", "POST"])
+def logout():
+    logout_user()
+    return redirect(url_for("home"))
+
+
+@app.route("/", methods=["GET", "POST"])
+def home():
+    return render_template("index.html")
+
+
+@app.route("/groups", methods=["GET", "POST"])
+@login_required
+def groups():
+    groups = (
+        db.session.query(Groups)
+        .join(UserGroup)
+        .filter(UserGroup.user_id == current_user.id)
+    )
+    form = UserGroupForm(request.form)
+    if request.method == "POST" and form.validate():
+        form.validate_group_id(group_id=form.group_id.data)
```

Comment:
Yikes, čia tokia klaiva klaidelė dėl kurios neleidžia prisijungti prie grupės. Tu į `validate_group_id` paduoti reiškmę, bet `validate_group_id` viduje bandai dar kartelį paimti data, lyg `group_id` būtų objektas. Plus dar reikia užtikrinti, kad jis būtų int, kad query veiktų teisingai.

Tai reikėtų arba čia pakeisti, arba `validate_group_id`.
```suggestion
        form.validate_group_id(group_id=form.group_id)
```

Reply:
Dėkui, dariau validation, kad nepridėtų grupės, jei jos dar nėra, ir matyt pritaisiau ir nepratestavau pakankamai :)